### PR TITLE
media-libs/lib{ggi,gii,ggigcp,ggimisc,ggiwmh}: fix HOMEPAGE, use https, EAPI7, improve ebuilds

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

This set of PRs fixes the libggi* and libgii packages:
* updates the HOMEPAGE and uses https
* updates the DESCRIPTION
* adds EAPI7 ebuild which:
   - doesn't install .la files
   - doesn't install .a files (disable-static)
   - fixes libgii to use mutlilib-minimal eclass (unfortunately i had to use multilib_copy_sources)
   - removes libggi directfb use flag since directfb is long gone

This also should fix following bugs:
https://bugs.gentoo.org/543040
https://bugs.gentoo.org/542622
https://bugs.gentoo.org/542612

Please review.

### CODE REVIEW

File: media-libs/libggi/libggi-2.2.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+HOMEPAGE="https://ibiblio.org/ggicore/packages/libggi.html"
+SRC_URI="mirror://sourceforge/ggi/${P}.src.tar.bz2"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd"
+IUSE="X aalib svga fbcon 3dfx debug cpu_flags_x86_mmx vis"
+
+RDEPEND=">=media-libs/libgii-1.0.2
+	X? ( x11-libs/libXt
```
Comment (a17r) [inline media-libs/libggi/libggi-2.2.2-r1.ebuild]: please keep them sorted

File: media-libs/libggi/libggi-2.2.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		x11-libs/libXxf86dga
+		x11-libs/libXxf86vm
+		x11-libs/libXt )
+	svga? ( >=media-libs/svgalib-1.4.2 )
+	aalib? ( >=media-libs/aalib-1.2-r1 )"
+DEPEND="${RDEPEND}
+	X? ( x11-base/xorg-proto )"
+
+DOCS="ChangeLog* FAQ NEWS README"
```
Comment (a17r) [inline media-libs/libggi/libggi-2.2.2-r1.ebuild]: make it an array

File: media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+RDEPEND=">=media-libs/libggi-2.2.2"
+DEPEND="${RDEPEND}"
+
+src_configure(){
+	econf --disable-static
+}
+
+src_install(){
+	DOCS=( ChangeLog README doc/{TODO,*.txt,*.faq} )
```
Comment (a17r) [inline media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild]: define it globally, after `DEPEND`

File: media-libs/libggimisc/libggimisc-2.2.2-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+src_configure() {
+	econf --disable-x --without-x \
+		$(use_enable svga svgalib) \
+		$(use_enable fbcon fbdev) \
+		--disable-static
+}
+
+src_install() {
+	DOCS=( ChangeLog README TODO doc/*.txt )
```
Comment (a17r) [inline media-libs/libggimisc/libggimisc-2.2.2-r1.ebuild]: define it globally

File: media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd"
+IUSE="X"
+
+RDEPEND=">=media-libs/libggi-2.2.2
+	X? ( x11-libs/libXxf86vm
+		x11-libs/libXxf86dga
+		x11-libs/libXext
```
Comment (a17r) [inline media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild]: keep them sorted

File: media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+src_configure() {
+	econf \
+		$(use_enable X x) \
+		$(use_with X x) \
+		--disable-static
+}
+
+src_install() {
+	DOCS=( ChangeLog README doc/*.txt )
```
Comment (a17r) [inline media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild]: define it globally

File: media-libs/libgii/libgii-1.0.2-r2.ebuild [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+HOMEPAGE="https://ibiblio.org/ggicore/packages/libgii.html"
+SRC_URI="mirror://sourceforge/ggi/${P}.src.tar.bz2"
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd"
+IUSE="X"
+
+RDEPEND="X? (
+	>=x11-libs/libX11-1.6.2[${MULTILIB_USEDEP}]
```
Comment (a17r) [inline media-libs/libgii/libgii-1.0.2-r2.ebuild]: please indent

File: media-libs/libggi/libggi-2.2.2-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+EAPI=7
+
+DESCRIPTION="Provides an opaque interface to the display's acceleration function"
+HOMEPAGE="https://ibiblio.org/ggicore/packages/libggi.html"
+SRC_URI="mirror://sourceforge/ggi/${P}.src.tar.bz2"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd"
+IUSE="X aalib svga fbcon 3dfx debug cpu_flags_x86_mmx vis"
```
Comment (a17r) [inline media-libs/libggi/libggi-2.2.2-r1.ebuild]: keep them sorted

File: media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="Windows manager hints extensions for libggi"
```
Comment (a17r) [inline media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild]: Probably should mean *Window* manager (singular)

File: media-libs/libggiwmh/libggiwmh-0.3.2.ebuild [hunk 10]
```diff
@@ -1,10 +1,10 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=0
 
-DESCRIPTION="Library for General Graphics Interface"
-HOMEPAGE="http://www.ggi-project.org"
+DESCRIPTION="Windows manager hints extensions for libggi"
```
Comment (a17r) [inline media-libs/libggiwmh/libggiwmh-0.3.2.ebuild]: Probably should mean *Window* manager (singular)

**Comment by mm1ke** [comment]
@a17r 
Thanks again for the review.
I've fixed almost all issue. However regarding the ```DOCS``` variable.
I can't define a ```DOCS``` array globally as the variable expansion wouldn't work there (and i would get an error when trying to test it). I can only use a normal variable or define the array in ```src_install```.
Even though i would prefer a array too, the variable should be fine too. PMS accepts them as far as i understand it.
Thus, i decided to make them globally and keep them as a variable where needed.

File: media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild (hunk 5 above)
Comment (mm1ke) [reply media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild]: don't know why this is still not outdated.. I've updated sorting ;)

**Comment by a17r** [comment]
> I can't define a `DOCS` array globally as the variable expansion wouldn't work there (and i would get an error when trying to test it). 

The solution here is to not use the expansion. ;)

**Comment by mm1ke** [comment]
> > I can't define a `DOCS` array globally as the variable expansion wouldn't work there (and i would get an error when trying to test it).
> 
> The solution here is to not use the expansion. ;)

Ah, ok sure... fixed now ;)

File: media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+SRC_URI="mirror://sourceforge/ggi/${P}.src.tar.bz2"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=media-libs/libggi-2.2.2"
+DEPEND="${RDEPEND}"
+
+DOCS=( ChangeLog README doc/TODO doc/libggigcp-functions.txt \
```
Comment (a17r) [inline media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild]: no need for trailing backslash in array, please remove

File: media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+	doc/libggigcp-libraries.txt doc/libggigcp-structures.txt \
+	doc/libggigcp.txt doc/colors.faq doc/colors2.faq )
```
Comment (a17r) [inline media-libs/libggigcp/libggigcp-1.0.2-r1.ebuild]: you can use `doc/{libggigcp{,-libraries,-structures,functions}}` `doc/colors{,2}.faq` to shorten the list

File: media-libs/libggimisc/libggimisc-2.2.2-r1.ebuild [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="fbcon svga"
+
+RDEPEND=">=media-libs/libggi-2.2.2
+	svga? ( media-libs/svgalib )"
+DEPEND="${RDEPEND}"
+
+DOCS=( ChangeLog README TODO doc/ggimisc.txt doc/libggimisc-functions.txt \
+	doc/libggimisc-libraries.txt doc/libggimisc.txt doc/retrace.txt )
```
Comment (a17r) [inline media-libs/libggimisc/libggimisc-2.2.2-r1.ebuild]: same as above

File: media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+RDEPEND=">=media-libs/libggi-2.2.2
+	X? (
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXxf86dga
+		x11-libs/libXxf86vm
+	)"
+DEPEND="${RDEPEND}"
+
+DOCS=( ChangeLog README doc/libggiwmh-functions.txt \
```
Comment (a17r) [inline media-libs/libggiwmh/libggiwmh-0.3.2-r1.ebuild]: same as above, and the line is not long enough to justify the line break anyway

**Comment by mm1ke** [comment]
@a17r 
Fixed, thanks again for reviewing.

**Pull request closed without merging**